src/ast.c

```diff
@@ -1046,6 +1046,125 @@ int has_meta(jl_array_t *body, jl_sym_t *sym)
     return 0;
 }
 
+void renumber_slots(jl_array_t *code, size_t *newslot);
+
+jl_lambda_info_t *desplat_lambda(jl_lambda_info_t *li, jl_tupletype_t *types)
+{
+    assert(li->isva);
+    size_t *newslot = (size_t*) malloc(jl_array_len(li->slotnames)*sizeof(size_t)+1);
```

Comment:
I think what you need is sth like

```
jl_array_t *root = NULL;
size_t *newslot;
size_t size = jl_array_len(li->slotnames) * sizeof(size_t) + 1; // what's the +1 for btw?
if (size > some_threshold) {
    root = jl_alloc_array_1d(jl_array_uint8_type, size);
    newslot = (size_t*)root->data;
}
else {
    newslot = (size_t*)alloca(size);
}
JL_GC_PUSH1(&root);
```

And of course merge the GC push and possible use a `JL_GC_PUSHARGS` instead...